constraints.txt

```diff
@@ -1 +1,5 @@
-# There are currently no constraints for Qiskit Algorithms
+# Numpy 1.25 deprecated some behaviours that we used, and caused the isometry
+# tests to flake. See https://github.com/Qiskit/qiskit-terra/issues/10305,
+# remove pin when resolving that.
+numpy<1.25
```

Comment:
Is this needed for the unit tests here - while terra may need to constrain things for the unit tests there I imagine it all works ok here or not? If not maybe it needs a bug being raised to fix things.

Reply:
Some unit tests were failing, I don't remember if they were from gradients or optimizers, but this is why I added the constraint.

Reply:
Ok, we'll have to look into it further then I guess to sort things once this merged.